src/test/surefire.go

```diff
@@ -0,0 +1,38 @@
+package test
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"core"
+	"fs"
+)
+
+func CopySurefireXmlFilesToDir(graph *core.BuildGraph, surefireDir string) {
+	outputDirs := make(map[string]struct{})
+	for _, target := range graph.AllTargets() {
+		if target.IsTest {
```

Comment:
probably worth checking `&& !target.NoTestOutput` here as well; that indicates that the target isn't expected to produce any test output.